src/react/components/pages/editorPage/canvas.tsx

```diff
@@ -329,9 +329,10 @@ export default class Canvas extends React.Component<ICanvasProps, ICanvasState>
 
```

Comment:
applyTag is calling updateRegions which calls async onAssetMetadataChanged. So when we call setTags, it is passed the project.tags before the tags have updated

Ideally we would pass the updated tags to setTags. This is a workaround